planner/core/rule_join_reorder.go

```diff
@@ -25,25 +25,59 @@ import (
 // construct a new join order based on a reorder algorithm.
 //
 // For example: "InnerJoin(InnerJoin(a, b), LeftJoin(c, d))"
-// results in a join group {a, b, LeftJoin(c, d)}.
```

Comment:
Since we also collect the left join. Point out how we maintain the outer join's information from the function's output. Also we can use a more complex example like `IJ{IJ{a, b}, RJ{LJ{c, d}, e}}`